src/lib/cookies/cookiemanager.cpp

```diff
@@ -137,9 +141,17 @@ void CookieManager::removeCookie()
         if (parentItem->childCount() == 0) {
             ui->cookieTree->deleteItem(parentItem);
         }
+        domain = cookie.domain();
     }
 
     mApp->cookieJar()->setAllCookies(allCookies);
+    return domain;
+}
+
+void CookieManager::blockDomain()
```

Comment:
This looks confusing. Blocking domain should not mean to remove current cookie + add domain to blacklist. It should just add cookie to blacklist.